File: work/inspect_voucher.py
import json
from pathlib import Path

import pandas as pd

path = Path('/Users/sunyitong/coding/voucher/代码资料/凭证信息_temp - 2026-06-02T140354.252.xlsx')
book = pd.ExcelFile(path)
print(json.dumps({'sheet_names': book.sheet_names}, ensure_ascii=False, indent=2))
for sheet in book.sheet_names:
    df = pd.read_excel(path, sheet_name=sheet, nrows=12)
    print(f'\nSHEET: {sheet}')
    print('columns:', json.dumps([str(c) for c in df.columns], ensure_ascii=False))
    print(df.head(8).to_string(index=False))

File: work/inspect_voucher.mjs
import { FileBlob, SpreadsheetFile } from "@oai/artifact-tool";

const inputPath = "/Users/sunyitong/coding/voucher/代码资料/凭证信息_temp - 2026-06-02T140354.252.xlsx";
const input = await FileBlob.load(inputPath);
const workbook = await SpreadsheetFile.importXlsx(input);

console.log(JSON.stringify({ sheetNames: workbook.worksheets.map(ws => ws.name) }, null, 2));

for (const ws of workbook.worksheets) {
  const used = ws.usedRange?.address ?? "unknown";
  console.log(`SHEET ${ws.name} USED ${used}`);
  const previewRange = `${ws.name}!A1:Z12`;
  const preview = await workbook.inspect({
    kind: "table",
    range: previewRange,
    include: "values,formulas",
    tableMaxRows: 12,
    tableMaxCols: 26,
  });
  console.log(preview.ndjson);
}
